src/engraving/layout/layoutsystem.cpp

```diff
@@ -107,7 +107,9 @@ System* LayoutSystem::collectSystem(const LayoutOptions& options, LayoutContext&
     while (ctx.curMeasure) {      // collect measure for system
         oldSystem = ctx.curMeasure->system();
         system->appendMeasure(ctx.curMeasure);
-
+        if (system->hasCrossStaffOrModifiedBeams()) {
```

Comment:
It looks like system->hasCrossStaffOrModifiedBeams() is a somewhat expensive operation. Would it be possible to call it only once, outside the loop?

Reply:
Quick answer is no, because every loop adds one more measure, which may add cross-staff or modified beams to the system, so I need to re-check. I may try to make it more efficient by only looking at the last measure when it's added, but that may cause issues if beams are connected across measures. I'll look into it.